tsdef/index.d.ts

```diff
@@ -10,7 +10,7 @@ import { Room } from './Room';
 export const isSupported: boolean;
 export const version:string;
 export const Logger: Log.RootLogger;
-export function connect(token: string, options?: ConnectOptions): Promise<Room>;
+export function connect(token: string, options?: ConnectOptions): CancelablePromise<Room>;
```

Comment:
is the definition of CancelablePromise public? Shouldn't it be imported in this file?

Reply:
Please  include definition of cancellable promise like:
```
interface CancelablePromise<T> extends Promise<T> {
  cancel: () => void
}
```
also include the usage in tests.

Reply:
I was thinking of actually writing up the CancelablePromise type def but I wasn't sure since it's not public. I noticed that our CancelablePromise isn't actually extending the class `Promise` as well.

Reply:
good point about `CancelablePromise` not extending from `Promise`.  In that case we can add its definition and exclude finally. But It would be better if we supported finally (and really extended it from Promise) - Maybe this should be part of convert `connect` to typescript work